### ISSUE TITLE
Column SetVisibleOutlineLevel and CollapseChildren methods don't work properly

### ISSUE THREAD

**Issue by xeniorn** [EPPlusSoftware/EPPlus#1095]
### EPPlus usage

Noncommercial use

### Environment

Windows 10

### Epplus version

6.2.10

### Spreadsheet application

Excel

### Description

The result is inconsistent and unstable and depends on the row contents.

Issue most likely comes from the following function:
{{ from ExcelRangeColumn.cs }}
```
private int GetLastCol()
{
	int num = ((_worksheet.Dimension != null) ? Math.Max(_worksheet.Dimension.End.Row, _worksheet._values.GetLastRow(0)) : _worksheet._values.GetLastColumn());
	if (_toCol <= num + 1)
	{
		return _toCol;
	}
	return num + 1;
}
```

Which was probably wrongly copied over from here:
{{ from ExcelRangeRow.cs }}
```
private int GetToRow()
{
	int num = ((_worksheet.Dimension != null) ? Math.Max(_worksheet.Dimension.End.Row, _worksheet._values.GetLastRow(0)) : _worksheet._values.GetLastRow(0));
	if (_toRow <= num + 1)
	{
		return _toRow;
	}
	return num + 1;
}
```

Test code:
1) try to set outline level, put data into the column, and set outline visibility in different order
2) repeat test N times, but adding some value in k-th row of column 1.

Results:
1) collapsing only works if there's a value in a row# that's >= of the column# currently being collapsed
2) collapsing otherwise works regardless of order, except that outline levels have to be set before outline level visibility is set

```
internal class Program
{

    
    static void Main(string[] args)
    {
        ExcelPackage.LicenseContext = LicenseContext.NonCommercial;
        var package = new ExcelPackage();
        var workbook = package.Workbook;

        

        string testName;
        int startColumn;
        int subColumnCount;
        bool doTest;

        

        for (var cycle = 1; cycle < 15; cycle++)
        {
            var ws = workbook.Worksheets.Add("test" + cycle);
            ws.OutLineSummaryRight = false;
            ws.Rows[0].Style.TextRotation = 90;
            ws.Row(1).CustomHeight = false;

            ws.ClearFormulas();
            ws.ClearFormulaValues();

            ws.Cells[cycle, 1].Value = "set";

            (testName, startColumn, subColumnCount, doTest) = ("outline-collapse[all]-data", 1, 3, true);
            if (doTest)
            {
                for (int i = 1; i <= subColumnCount; i++)
                {
                    ws.Column(startColumn + i).OutlineLevel = i;
                }
                for (int i = 0; i <= subColumnCount; i++)
                {
                    ws.Cells[1, startColumn + i].EntireColumn.SetVisibleOutlineLevel(0);
                }
                for (int i = 0; i <= subColumnCount; i++)
                {
                    ws.Cells[1, startColumn + i].Value = testName + i;
                }
            }

            (testName, startColumn, subColumnCount, doTest) = ("outline-collapse[first]-data", 5, 3, true);
            if (doTest)
            {
                for (int i = 1; i <= subColumnCount; i++)
                {
                    ws.Column(startColumn + i).OutlineLevel = i;
                }
                for (int i = 0; i <= subColumnCount; i++)
                {
                    ws.Cells[1, startColumn + i].EntireColumn.SetVisibleOutlineLevel(0);
                }
                for (int i = 0; i <= subColumnCount; i++)
                {
                    ws.Cells[1, startColumn + i].Value = testName + i;
                }
            }

            (testName, startColumn, subColumnCount, doTest) = ("collapse[first]-outline-data", 9, 3, true);
            if (doTest)
            {
                for (int i = 0; i <= 0; i++)
                {
                    ws.Cells[1, startColumn + i].EntireColumn.SetVisibleOutlineLevel(0);
                }
                for (int i = 1; i <= subColumnCount; i++)
                {
                    ws.Column(startColumn + i).OutlineLevel = i;
                }
                for (int i = 0; i <= subColumnCount; i++)
                {
                    ws.Cells[1, startColumn + i].Value = testName + i;
                }
            }

            (testName, startColumn, subColumnCount, doTest) = ("data-outline-collapse[first]", 13, 3, true);
            if (doTest)
            {
                for (int i = 0; i <= subColumnCount; i++)
                {
                    ws.Cells[1, startColumn + i].Value = testName + i;
                }
                for (int i = 1; i <= subColumnCount; i++)
                {
                    ws.Column(startColumn + i).OutlineLevel = i;
                }
                for (int i = 0; i <= 0; i++)
                {
                    ws.Cells[1, startColumn + i].EntireColumn.SetVisibleOutlineLevel(0);
                }
            }

            var s = "";
            for (var i = 1; i <= 50; i++)
            {
                var column = ws.Column(i);
                
                if (column is null) { continue; }
                
                s = s + (column.Collapsed ? "1" : 0);
            }
            Console.WriteLine($"cyc{cycle}::{s}");

        }

        var path = @"c:\temp\testxel\";
        var name = Guid.NewGuid().ToString();
        System.IO.Directory.CreateDirectory(path);
        var file = Path.Combine(path, $"{name}.xlsx");
        package.SaveAs(file);
        var p = new Process()
        {
            StartInfo = new ProcessStartInfo(file) { UseShellExecute = true },
        };
        p.Start();

    }
       
}
```

Behavior same in versions 6.1 - 7.0.0-rc.

Cheers,

xen

**Comment by JanKallman** [comment]
If the last column did not have any values, the GetLastCol() method looked incorrectly returned the last Row with values instead of the column. I will investigate some more and add a fix for this.

**Comment by JanKallman** [comment]
Fixed in 6.2.12 and  7.0.1
